source/ArchivedMessages.Infrastructure/MeteringPointQueryBuilder.cs

```diff
@@ -0,0 +1,176 @@
+﻿// Copyright 2020 Energinet DataHub A/S
+//
+// Licensed under the Apache License, Version 2.0 (the "License2");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Dapper;
+using Energinet.DataHub.EDI.ArchivedMessages.Domain.Models;
+using Energinet.DataHub.EDI.BuildingBlocks.Domain.Authentication;
+using Energinet.DataHub.EDI.BuildingBlocks.Domain.Exceptions;
+using Energinet.DataHub.EDI.BuildingBlocks.Domain.Models;
+
+namespace Energinet.DataHub.EDI.ArchivedMessages.Infrastructure;
+
+internal sealed class MeteringPointQueryBuilder(ActorIdentity actorIdentity)
+{
+    private readonly DynamicParameters _queryParameters = new();
+    private readonly List<string> _statement = new();
+    private readonly ActorIdentity _actorIdentity = actorIdentity;
+
+    /// <summary>
+    /// Build a query for fetching archived messages based on the given query options.
+    /// </summary>
+    /// <param name="query"></param>
+    public QueryInput BuildFrom(GetMeteringPointMessagesQuery query)
+    {
+        if (query.MeteringPointId is null)
+            throw new ArgumentNullException(nameof(query.MeteringPointId), "MeteringPointId cannot be null.");
+        if (query.CreationPeriod is null)
+            throw new ArgumentNullException(nameof(query.CreationPeriod), "CreationPeriod cannot be null.");
+
+        AddFilter(
+            "JSON_QUERY(MeteringPointIds) IS NOT NULL AND EXISTS (SELECT 1 FROM OPENJSON(MeteringPointIds)  WHERE value = @MeteringPointId)",
+            new KeyValuePair<string, object>("MeteringPointId", query.MeteringPointId.Value));
+
+        AddFilter(
+            "CreatedAt BETWEEN @StartOfPeriod AND @EndOfPeriod",
+            new KeyValuePair<string, object>("StartOfPeriod", query.CreationPeriod.DateToSearchFrom.ToString()),
+            new KeyValuePair<string, object>("EndOfPeriod", query.CreationPeriod.DateToSearchTo.ToString()));
+
+        if (query.Sender is not null && query.Receiver is not null)
+        {
+            AddFilter(
+                "(SenderNumber=@SenderNumber OR ReceiverNumber=@ReceiverNumber)",
+                new KeyValuePair<string, object>("SenderNumber", query.Sender.ActorNumber.Value),
+                new KeyValuePair<string, object>("ReceiverNumber", query.Receiver.ActorNumber.Value));
+
+            AddFilter(
+                "(SenderRoleCode=@SenderRoleCode OR ReceiverRoleCode=@ReceiverRoleCode)",
```

Comment:
```suggestion
                "(SenderRoleCode=@SenderRoleCode OR ReceiverRoleCode=@ReceiverRoleCode)",
```
We should remove the "Code", since we're not storing a code anymore